#175 - Optimized, common selection algorithm for all navigation methods

tfesenko opened the pull request:
![swagger_selection](https://cloud.githubusercontent.com/assets/644582/21859696/adb2ef1c-d7fb-11e6-9877-7cdcd519666f.gif)

tfesenko committed:
#175 - Optimized, common selection algorithm for all navigation methods

andylowry commented on com.reprezen.swagedit/src/com/reprezen/swagedit/model/AbstractNode.java:
```diff
@@ -175,7 +176,11 @@ public Position getPosition(IDocument document) {
+            if (selectEntireElement) {
+                length = (document.getLineOffset(endLine) + endCol) - offset;
+            } else {
+                length = document.getLineOffset(startLine+1) - offset;
```
Coulld this throw BadLocationException if startLine is the last line in the document?

andylowry reviewed (commented):
@tfesenko One question for you to consider and handle as needed. Otherwise looks good.

andylowry committed:
175 - Handle edge case in setting selection

If an object is selected and that object is fully contained in the final
line of the file and does not have a line terminator, the code for
computing the range to be highlighed can't ask for the position of the
next line without throwing BadLocationException. That case is now
checked for explicitly.
